05.ls/my_ls4.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+require 'optparse'
+require 'etc'
+require 'date'
+
+NUMBER_OF_COLUMN = 3
+NUMBER_OF_SPACE = 4
+
+def main
+  options = ARGV.getopts('l')
+  files = Dir.glob('*').sort
+  if options['l']
+    l_option(files)
+  else
+    other_than_l_option(files)
+  end
+end
+
+def l_option(files)
+  display_total_block_size(files)
+  display_files_information(files)
+end
+
+def display_total_block_size(files)
+  total_block_size = 0
+  files.each do |file|
+    total_block_size += File.lstat(file).blocks
+  end
+  puts "total #{total_block_size}"
+end
+
+def display_files_information(files)
+  files.map do |file|
+    stat = File.lstat(file)
+
+    filetype_short = make_filetype(stat.ftype)
+
+    filemode = stat.mode.to_s(8).rjust(6, '0')
+    filemode_owner = make_filemode(filemode[3].to_i)
+    filemode_owned_group = make_filemode(filemode[4].to_i)
+    filemode_other_users = make_filemode(filemode[5].to_i)
+    permissions = "#{filemode_owner}#{filemode_owned_group}#{filemode_other_users}" # 式展開が長くなりすぎるのでまとめた
+
+    hardlink = stat.nlink
+    owner_name = Etc.getpwuid(stat.uid).name
+    group_name = Etc.getgrgid(stat.gid).name
+
+    filesize = stat.size.to_s.rjust(make_length_of_max_size_file(files))
+
+    month = stat.mtime.strftime('%m').to_i.to_s.rjust(2)
+    day = stat.mtime.strftime('%e')
+    time = stat.mtime.strftime('%H:%M')
+    time_stamp = "#{month} #{day} #{time}" # 式展開が長くなりすぎるのでまとめた
+
+    path = File.symlink?(file)
+    link_source_of_symbolic_link = " -> #{File.realpath(file).split('/')[-1]}" if path
```

Comment:
`path`という変数名はなんか変ですね。文字通りpathが入っていそうですが、実際には`true/false`が入っているんですよね。`is_symlink`とかが妥当な気がします。もっと言えば、変数に入れる意味もあまりなさそうですので下記のようにするのが良いんじゃないでしょうか。

```suggestion
    link_source_of_symbolic_link = " -> #{File.realpath(file).split('/')[-1]}" if File.symlink?(file)
```